packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/scheduling/NotificationScheduler.java

```diff
@@ -0,0 +1,118 @@
+package expo.modules.notifications.notifications.scheduling;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.os.Handler;
+import android.os.Looper;
+import android.os.ResultReceiver;
+
+import org.json.JSONObject;
+import org.unimodules.core.ExportedModule;
+import org.unimodules.core.Promise;
+import org.unimodules.core.arguments.ReadableArguments;
+import org.unimodules.core.errors.InvalidArgumentException;
+import org.unimodules.core.interfaces.ExpoMethod;
+
+import java.util.Map;
+
+import androidx.annotation.Nullable;
+import expo.modules.notifications.notifications.interfaces.SchedulableNotificationTrigger;
+import expo.modules.notifications.notifications.service.ExpoNotificationSchedulerService;
+import expo.modules.notifications.notifications.triggers.DateTrigger;
+import expo.modules.notifications.notifications.triggers.TimeIntervalTrigger;
+
+public class NotificationScheduler extends ExportedModule {
+  private final static String EXPORTED_NAME = "ExpoNotificationScheduler";
+
+  private static Handler HANDLER = new Handler(Looper.getMainLooper());
+
+  public NotificationScheduler(Context context) {
+    super(context);
+  }
+
+  @Override
+  public String getName() {
+    return EXPORTED_NAME;
+  }
+
+  @ExpoMethod
+  public void getAllScheduledNotificationsAsync(final Promise promise) {
+    ExpoNotificationSchedulerService.enqueueFetchAll(getContext(), new ResultReceiver(HANDLER) {
+      @Override
+      protected void onReceiveResult(int resultCode, Bundle resultData) {
+        super.onReceiveResult(resultCode, resultData);
+        if (resultCode == ExpoNotificationSchedulerService.SUCCESS_CODE) {
+          promise.resolve(resultData.getParcelableArrayList(ExpoNotificationSchedulerService.NOTIFICATIONS_KEY));
+        } else {
+          Exception e = resultData.getParcelable(ExpoNotificationSchedulerService.EXCEPTION_KEY);
+          promise.reject("ERR_FAILED_TO_FETCH", "Failed to fetch scheduled notifications.", e);
```

Comment:
```suggestion
          promise.reject("ERR_NOTIFICATIONS_FAILED_TO_FETCH", "Failed to fetch scheduled notifications.", e);
```
and similar below (should include the module name in error codes)